src/Controller/JobAdvertsController.php

```diff
@@ -0,0 +1,107 @@
+            ->then(function (Sequence $sequence) use ($page, $perPage) {
+                $pagerfanta = new Pagerfanta(new SequenceAdapter($sequence, $this->willConvertTo(Teaser::class)));
+                $pagerfanta->setMaxPerPage($perPage)->setCurrentPage($page);
+
+                return $pagerfanta;
+            });
+
+        $arguments['title'] = 'eLife Jobs';
+
+        $arguments['paginator'] = $latest
+            ->then(function (Pagerfanta $pagerfanta) use ($request) {
+                return new Paginator(
+                    'Browse our open job adverts',
+                    $pagerfanta,
+                    function (int $page = null) use ($request) {
+                        $routeParams = $request->attributes->get('_route_params');
+                        $routeParams['page'] = $page;
+
+                        return $this->get('router')->generate('job-adverts', $routeParams);
+                    }
+                );
+            });
+
+        $arguments['listing'] = $arguments['paginator']
+            ->then($this->willConvertTo(ListingTeasers::class, ['heading' => 'Latest', 'type' => 'job-adverts', 'emptyText' => 'No vacancies at present.  Feel free to send your CV to hr@elifesciences.org']));
+
+        if (1 === $page) {
+            return $this->createFirstPage($arguments);
+        }
+
+        return $this->createSubsequentPage($request, $arguments);
+    }
+
+    private function createFirstPage(array $arguments) : Response
+    {
+        $arguments['contentHeader'] = new ContentHeader(
+          $arguments['title'],
+          null
+        );
+
+        return new Response($this->get('templating')->render('::job-adverts.html.twig', $arguments));
+    }
+
+    public function jobAdvertAction(Request $request, string $id) : Response
+    {
+        $jobAdvert = $this->get('elife.api_sdk.job_adverts')
+            ->get($id)
+            ->otherwise($this->mightNotExist())
+            ->then(function (JobAdvert $jobAdvert) {
+                return $jobAdvert;
+            })
+            ->then($this->checkSlug($request, Callback::method('getTitle')));
+
+        $arguments = $this->defaultPageArguments($request, $jobAdvert);
+
+        $arguments['title'] = $jobAdvert
+            ->then(Callback::method('getTitle'));
+
+        $arguments['job-advert'] = $jobAdvert;
+
+        $arguments['contentHeader'] = $arguments['job-advert']
+            ->then($this->willConvertTo(ContentHeader::class));
+
+        $arguments['blocks'] = $arguments['job-advert']
+            ->then(function (JobAdvert $jobAdvert) {
+
+              if ($jobAdvert->getClosingDate() > new \DateTimeImmutable('now')) {
+                  return $this->convertContent($jobAdvert)
+                    ->prepend(new Paragraph("Closing date for application is  <b>{$jobAdvert->getClosingDate()->format('M d Y')}</b>"));
+                }
+
+                return [new Paragraph('This position is now closed to applications.')];
```

Comment:
Maybe we should return a `410 Gone` to delist from search indexes? We _don't_ do that for events though.

Reply:
I like that idea. Where would that happen?

Reply:
Throw a `GoneHttpException`. That won't allow for a custom page though. Maybe we can add that later as part of the error pattern change.

Alternatively, throw a `EarlyResponse` (hacky as exception as flow control) with your page and status code.